pkg/internal/cluster/create/actions/config/config.go

```diff
@@ -153,12 +165,21 @@ func removeMetadata(kustomized string) string {
 	)
 }
 
-func allPatchesFromConfig(cfg *config.Cluster) (patches []string, jsonPatches []config.PatchJSON6902) {
-	return cfg.KubeadmConfigPatches, cfg.KubeadmConfigPatchesJSON6902
+func allPatchesFromConfig(cfg *config.Cluster, includeWorkerPatches bool) (patches []string, jsonPatches []config.PatchJSON6902) {
+	patches, jsonPatches = cfg.KubeadmConfigPatches, cfg.KubeadmConfigPatchesJSON6902
+
+	if includeWorkerPatches {
+		for _, node := range cfg.Nodes {
```

Comment:
Why would ever node's patches be applied to all nodes? Why not just the patch for the current node?

Reply:
Yes, you're right. The patches should be applied to the current node. Looks like we would need to send the current node down to this function or the like.